app/src/components/DetailsBoard.js

```diff
@@ -89,16 +97,192 @@ function DetailsBoard({
+        orderBy: "startTime",
+      })
+      .then(function (response) {
+        var busyArr = response.result.items;
+
+        var freeArr = [];
+
+        var busyReducer = (freeTime, busyTime) => {
+          freeTime.end = busyTime.start;
+          freeArr.push({ freeTime });
+
+          return { start: busyTime.end };
+        };
+
+        busyArr.reduce(busyReducer, {
+          start: {
+            dateTime: currentDate.toISOString(),
+            timeZone: "America/Los_Angeles",
+          },
+        });
+
+        if (freeArr.length > 0) {
+          return freeArr.map((freeTime, index) => {
+            var { freeTime } = freeTime;
+            var totalTime =
+              new Date(freeTime.end.dateTime).getTime() -
+              new Date(freeTime.start.dateTime).getTime();
+
+            freeTime.totalTime = totalTime;
+
+            return freeTime;
+          });
+        } else {
+          return ["No free times. :["];
+        }
+      });
+
+    setFreeTimesRemaining(timeBlocks);
+  }
+
+  //reduce free times to total free time
+  function getSumFreeTime() {
+    //console.debug("getSumFreeTime()"); //Test
+
+    function getMilliseconds(total, current) {
+      let addedTime = current.totalTime < 0 ? 0 : current.totalTime;
+      return total + addedTime;
+    }
+
+    let msHours = 1000 * 60 * 60;
+    let msMinutes = 1000 * 60;
+
+    let totalMS = freeTimesRemaining.reduce(getMilliseconds, 0);
+    let hours = totalMS / msHours;
+    let minutes = (totalMS % msHours) / msMinutes;
+
+    setTotalFreeTime({ totalMS: totalMS, hours: hours, minutes: minutes });
+  }
+
+  //get proportion of each free time block
+  function getFreeTimePercentages() {
+    let update = freeTimesRemaining.map((freetime) => {
+      freetime.freeTimePercentage =
+        parseInt(freetime.totalTime) / totalFreeTime.totalMS;
+      return freetime;
+    });
+
+    setFreeTimesRemaining(update);
+
+    getSuggestions();
+  }
+
+  //get suggested time spent working on task
+  function getSuggestions() {
+    let task_hoursToMS = testTask.estTime.hours * 60 * 60 * 1000;
+    let msHours = 1000 * 60 * 60;
+    let msMinutes = 1000 * 60;
+
+    let totals = freeTimesRemaining.map((time) => {
+      let amount = time.freeTimePercentage * task_hoursToMS;
+      let hours = Math.round(amount / msHours);
+      let minutes = Math.round((amount % msHours) / msMinutes);
+
+      return { totalMS: amount, hours, minutes };
+    });
+
+    setSuggestions(totals);
+  }
+
+  //USE EFFECTS
+  //When component mounts, gets array of free timeblocks and total remaining time until due date
+  React.useEffect(async () => {
+    getRemainingFreeTimes(testTask.dueDate, currentDate);
+    setTimeRemaining(getRemainingTime(testTask.dueDate, currentDate));
+  }, []);
+
+  //When length of freeTimesRemaining changes (aka they've loaded in)
+  //Gets sum of all free time
+  React.useEffect(() => {
+    if (freeTimesRemaining.length !== 0) {
```

Comment:
I think you can get rid of the `if` check, since an empty `freeTimesRemaining` array should reduce to 0 (since the `.reduce` function is initialized to `0`).